superset-frontend/src/explore/components/controls/DatasourceControl/index.jsx

```diff
@@ -156,33 +198,33 @@ class DatasourceControl extends React.PureComponent {
     if (this.props.onDatasourceSave) {
       this.props.onDatasourceSave(datasource);
     }
-  }
+  };
 
-  toggleShowDatasource() {
+  toggleShowDatasource = () => {
     this.setState(({ showDatasource }) => ({
       showDatasource: !showDatasource,
```

Comment:
what happens if someone double clicks on the x sign with this logic? Wouldn't it be better to have it change it to false or true instead?

Reply:
the `!` is doing the toggle functionality logic wise